foreshadow/steps/preparerstep.py

```diff
@@ -263,12 +270,64 @@ class instead of normal Pipelines.  .. #noqa: I102
             **kwargs: kwargs to PIpeline constructor.
 
         """
+        self._parallel_process = None
+        if "_parallel_process" in kwargs:
+            self._parallel_process = kwargs.pop("_parallel_process")
         self.column_sharer = column_sharer
         if self.column_sharer is None:
             self.column_sharer = ColumnSharer()
-        self._parallel_process = None
         super().__init__(**kwargs)
 
+    def configure_column_sharer(self, column_sharer):  # noqa
+        self.column_sharer = column_sharer
+        self._parallel_process.configure_column_sharer(column_sharer)
+
+    def dict_serialize(self, deep=True):  # noqa
+        selected_params = self.get_params(deep=deep)["_parallel_process"]
+        serialized = _make_serializable(
+            selected_params, serialize_args=self.serialize_params
+        )
+        transformer_list = serialized["transformer_list"]
+        serialized.pop("transformer_list")
+        serialized["transformation_by_column_group"] = transformer_list
+        return serialized
+
+    @classmethod
+    def dict_deserialize(cls, data):  # noqa
+        params = _make_deserializable(data)
+        import pdb
+
+        pdb.set_trace()
+        parallel_processor = cls.reconstruct_parallel_process(params)
+        reconstructed_params = {"_parallel_process": parallel_processor}
+        ret_tf = cls()
+        ret_tf.set_params(**reconstructed_params)
+        return ret_tf
+
+    @classmethod
+    def reconstruct_parallel_process(cls, data):  # noqa
+        import pdb
+
+        pdb.set_trace()
+        n_jobs = data["n_jobs"]
+        transformer_weights = data["transformer_weights"]
+        collapse_index = data["collapse_index"]
+
+        transformer_list = []
+        for i, transformation in enumerate(
+            data["transformation_by_column_group"]
+        ):
+            group_name = "group: {}".format(str(i))
```

Comment:
Nice, this should mean we can create new groups dynamically in deserialization from the .json